src/hpr_config_worker.erl

```diff
@@ -0,0 +1,170 @@
+            {noreply, State, {continue, connect}}
+    end;
+handle_continue(
+    init_stream,
+    #state{
+        connection = Connection
+    } = State
+) ->
+    {ok, Stream} = grpc_client:new_stream(
+        Connection, 'helium.config.config_service', route_updates, client_config_pb
+    ),
+    %% Sending Route Request
+    RouteReq = #{},
+    ok = grpc_client:send(Stream, RouteReq),
+    lager:info("stream initialized"),
+    ok = cfg_update(),
+    {noreply, State#state{stream = Stream}}.
+
+handle_call(Msg, _From, State) ->
+    {stop, {unimplemented_call, Msg}, State}.
+
+handle_cast(Msg, State) ->
+    {stop, {unimplemented_cast, Msg}, State}.
+
+handle_info(?RCV_CFG_UPDATE, #state{connection = Connection, stream = Stream} = State) ->
+    case grpc_client:rcv(Stream, ?RCV_TIMEOUT) of
+        {headers, _Headers} ->
+            ok = cfg_update(),
+            {noreply, State};
+        {data, RoutesResV1} ->
+            lager:info("got router update"),
+            ok = process_routes_update(RoutesResV1, State),
+            ok = cfg_update(),
+            {noreply, State};
+        eof ->
+            lager:warning("got eof"),
+            _ = grpc_client:stop_connection(Connection),
+            {noreply, State, {continue, connect}};
+        {error, timeout} ->
+            ok = cfg_update(),
+            {noreply, State};
+        {error, E} ->
+            lager:error("failed to rcv ~p", [E]),
+            {stop, {error, E}}
+    end;
+handle_info(_Msg, State) ->
+    {noreply, State}.
+
+terminate(_Reason, #state{connection = Connection}) ->
+    lager:error("terminate ~p", [_Reason]),
+    _ = grpc_client:stop_connection(Connection),
+    ok.
+
+%% ------------------------------------------------------------------
+%% Internal Function Definitions
+%% ------------------------------------------------------------------
+
+-spec maybe_init_dets(#state{}) -> #state{}.
+maybe_init_dets(#state{dets_backup_path = undefined} = State) ->
+    State;
+maybe_init_dets(#state{dets_backup_path = Path} = State) ->
+    ok = filelib:ensure_dir(Path),
+    {ok, Dets} = dets:open_file(Path, [{type, set}]),
+    case dets:lookup(Dets, last_update) of
+        [{last_update, LastRoutesResV1}] ->
+            ok = hpr_config:update_routes(LastRoutesResV1),
+            ok;
+        [] ->
+            ok
+    end,
+    State#state{dets = Dets}.
+
+-spec cfg_update() -> ok.
+cfg_update() ->
+    self() ! ?RCV_CFG_UPDATE,
+    ok.
+
+-spec process_routes_update(routes_req_v1_map(), #state{}) -> ok.
+process_routes_update(RoutesResV1, State) ->
+    ok = hpr_config:update_routes(RoutesResV1),
+    maybe_cache_response(RoutesResV1, State),
+    ok.
+
+-spec maybe_cache_response(RoutesResV1 :: routes_req_v1_map(), #state{}) -> ok.
+maybe_cache_response(_RoutesResV1, #state{dets = undefined}) ->
+    ok;
+maybe_cache_response(RoutesResV1, #state{dets = Dets}) ->
+    dets:insert(Dets, {last_update, RoutesResV1}).
```

Comment:
The `dets` file only ever has one row. Can't we use `file:write_file/2` and `file:read_file/1` with `term_to_binary/1` and `binary_to_term/1` instead?